editor/plugins/canvas_item_editor_plugin.cpp

```diff
@@ -1035,6 +1035,29 @@ void CanvasItemEditor::_viewport_input_event(const InputEvent &p_event) {
 
 		const InputEventMouseButton &b = p_event.mouse_button;
 
+		if (b.button_index == GESTURE_MAGNIFY) {
+
+			if (zoom < MIN_ZOOM) {
```

Comment:
It has to be documented really well though. It's not obvious that you have to use the factor property to find out if the user scrolls up or down.